packages/webapp/src/containers/userFarmSlice.ts

```diff
@@ -1,46 +1,75 @@
-import { createSlice } from '@reduxjs/toolkit';
+// https://redux-toolkit.js.org/tutorials/typescript#define-slice-state-and-action-types
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { createSelector } from 'reselect';
+import type { RootState } from '../store/store';
+import { AxiosError } from 'axios';
 
-export function onLoadingStart(state) {
+export interface Units {
+  currency: string;
+  measurement: 'metric' | 'imperial';
+}
+
+interface UserFarm {
+  farm_id: string;
+  user_id: string;
+  units: Units;
+
+  // https://www.typescriptlang.org/glossary#index-signatures
+  [key: string]: any;
+}
```

Comment:
I looked a bit more into this and I think the index signature is still the best way to avoid having to write out the whole interface to `UserFarm` (a Partial would be correct as well, but would still require the base interface which I think we want to avoid in these situations where it's not relevant and we aren't trying to type check on it).

Reply:
I think this interface could go in your new types file as we scope our data hierarchy. I think its good to write out the whole interface but I also am good with the idea of leaving it for now and maybe using whatever we touch as candidates for the next .ts update. Like a contagion we can create a ticket and say we need to update userFarm to .ts next.

Reply:
Yeah, let's talk about that with @antsgar when she returns in terms of how we want to manage the transition over to TS. I would prefer to not write any interfaces that we aren't actively using in code (i.e. where we aren't actually type checking on the properties) because it feels like bloat. So I like the "whatever we touch" approach personally (like the full interface would be written the day we use the `userFarm` selector in a TS file and need the properties), but it might be easier to establish a system now rather than do piecemeal 🤔